packages/deployed-backend-client/src/deployed_backend_client.ts

```diff
@@ -83,22 +84,34 @@ export class DefaultDeployedBackendClient implements DeployedBackendClient {
   };
 
   /**
-   * Returns Amplify Sandboxes for the account and region. The number of sandboxes returned can vary
+   * Returns a list of stacks for specific deployment type and status
+   * @yields
    */
-  listSandboxes = async (
-    listSandboxesRequest?: ListSandboxesRequest
-  ): Promise<ListSandboxesResponse> => {
-    const stackMetadata: SandboxMetadata[] = [];
-    let nextToken = listSandboxesRequest?.nextToken;
-
+  async *listBackends(listBackendsRequest?: ListBackendsRequest) {
+    const stackMetadata: BackendSummaryMetadata[] = [];
+
+    let nextToken;
+    const deploymentType = listBackendsRequest?.deploymentType;
+    const statusFilter = listBackendsRequest?.backendStatusFilters
+      ? listBackendsRequest?.backendStatusFilters
+      : [];
     do {
-      const listStacksResponse = await this.listStacks(nextToken);
-      const stackMetadataPromises = listStacksResponse.stackSummaries
-        .filter((stackSummary: StackSummary) => {
-          return stackSummary.StackStatus !== StackStatus.DELETE_COMPLETE;
-        })
+      const listStacksResponse = await this.listStacks(nextToken, statusFilter);
+
+      const listStacksSummaries =
+        deploymentType === 'sandbox'
+          ? listStacksResponse.stackSummaries.filter(
+              (stackSummary: StackSummary) => {
+                return stackSummary.StackStatus !== StackStatus.DELETE_COMPLETE;
+              }
+            )
+          : listStacksResponse.stackSummaries;
```

Comment:
We shouldn't filter statuses for one particular deployment type if the caller didn't ask it. Especially now that we have a generic method with a stack status as input.

Reply:
This a not(!) filter where we are excluding the stacks with 'DELETE_COMPLETE' status. If we don't add this filter here, we'll have to pass an array of all other(more than 20) statuses in the listStacks() call

Reply:
I think it makes sense to exclude DELETE_COMPLETE unless the filter explicitly requests it. But we should do it in the listStacks command, not as a client side filter. You can get all of the statuses except this one with the following code
```ts
StackStatusFilter: Object.values(StackStatus).filter(status => status !== StackStatus.DELETE_COMPLETE)
```